src/coreclr/src/vm/castcache.h

```diff
@@ -262,20 +262,20 @@ class CastCache
         // then we use fibonacci hashing to reduce the value to desired size.
 
 #if BIT64
-        TADDR hash = (((ULONGLONG)source << 32) | ((ULONGLONG)source >> 32)) ^ target;
+        UINT64 hash = (((UINT64)source << 32) | ((UINT64)source >> 32)) ^ (UINT64)target;
```

Comment:
Just if you wonder. The type changes like this are mostly to make C++ implementation as similar to C# version as possible.